### PR TITLE
Permissions and User API Structure

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR focuses on fleshing out the anticipated API structure and base code structure for supporting permission management and user discovery.

[SHOWCASE-2337](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-2337)
[SHOWCASE-2359](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-2359)

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by TimCsaky — APPROVED** [review]

File: app/src/controllers/permission.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+const controller = {
+  /** Searches for object permissions */
+  // eslint-disable-next-line no-unused-vars
+  objectPermissionSearch(req, res, next) {
+    new Problem(501).send(res);
+  },
+
+  /** Returns the object permissions */
+  // eslint-disable-next-line no-unused-vars
+  objectPermission(req, res, next) {
```
Comment (MichaelChung123) [inline app/src/controllers/permission.js:17]: should this be named "objectPermissions"?

File: app/src/controllers/permission.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    new Problem(501).send(res);
+  },
+
+  /** Grants object permissions to a specific user */
+  async addPermission(req, res, next) {
```
Comment (MichaelChung123) [inline app/src/controllers/permission.js:22]: should this be named "addPermissions"?

File: app/src/controllers/permission.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      const response = await recordService.share(req);
+      res.status(201).json(response);
+    } catch (e) {
+      next(errorToProblem(SERVICE, e));
+    }
+  },
+
+  /** Deletes object permissions for a specific user */
+  // eslint-disable-next-line no-unused-vars
+  removePermission(req, res, next) {
```
Comment (MichaelChung123) [inline app/src/controllers/permission.js:33]: should this be named "removePermissions"?

**Review by MichaelChung123 — COMMENTED** [review]
Few minor naming changes, otherwise looks good

**Pull request merged**